File: src/main/kotlin/network/bobnet/cms/service/administration/LogService.kt
package network.bobnet.cms.service.administration

import org.slf4j.Logger
import org.slf4j.LoggerFactory

class LogService(val name: Class<Any>){
    val logger: Logger = LoggerFactory.getLogger(name)

    fun info(message: String){
        logger.info(message)
    }

    fun warn(message: String){
        logger.warn(message)
    }

    fun error(message: String){
        logger.error(message)
    }

    fun debug(message: String){
        logger.debug(message)
    }
}

File: backend/src/service/logService.ts
import { configure, getLogger } from "log4js";
// @ts-ignore
import * as Config from "../config/config";

const logger = getLogger();
logger.level = Config.logLevel;

export const trace = (msg) => {
    logger.trace(msg);
}

export const debug = (msg) => {
    logger.debug(msg);
}

export const info = (msg) => {
    logger.info(msg);
}

export const warn = (msg) => {
    logger.warn(msg);
}

export const error = (msg) => {
    logger.error(msg);
}

export const fatal = (msg) => {
    logger.fatal(msg);
}